app/controllers/api/v1/users_controller.rb

```diff
@@ -7,13 +7,18 @@ class Api::V1::UsersController < Api::ApiController
   param :q, String, required: false, desc: "Query string"
   param :match, String, required: false, desc: "If set to 'exact', requires exact username match on q instead of prefix match"
   param :page, :number, required: false, desc: 'Page in results (25 per page)'
+  param :hide_unblockable, ['true', 'false'], required: false, desc: "If set to 'true', elimiates users who cannot be blocked from the result"
   error 422, "Invalid parameters provided"
   def index
     queryset = if params[:match] == 'exact'
       User.where(username: params[:q])
     else
       User.where("username LIKE ?", params[:q].to_s + '%').ordered
     end
+    if params[:hide_unblockable].present? && params[:hide_unblockable] == 'true'
```

Comment:
technically might require a logged_in? check but this is not urgent cause this flow shouldn't happen except for logged in users so it's just error hardening the API against bad users. Historical note that including API examples of this wouldn't be helpful because one of APIPie's current gaps is that it doesn't show the _request_ format.